src/pages/Wishlist/components/WishlistCard.js

```diff
@@ -0,0 +1,75 @@
+import { useData } from '../../../context';
+import styles from '../../Products/components/ProductCard.module.css';
+import FavoriteBorderIcon from '@material-ui/icons/FavoriteBorder';
+import FavoriteIcon from '@material-ui/icons/Favorite';
+import { useNavigate } from 'react-router';
+import axios from 'axios';
+
+export default function WishlistCard({ wishlistItem }) {
+  const { _id: id, image, name, price } = wishlistItem;
+  const { dataDispatch, wishlist, cart } = useData();
+  const isInCart = cart.find((cartItem) => cartItem._id === id);
+  const isInWishlist = wishlist.find((wishlistItem) => wishlistItem._id === id);
+  const navigate = useNavigate();
+
+  const addToCart = async (id) => {
+    const userID = 124;
+    try {
+      const response = await axios.post(`https://shoemeup.pranshudobhal.repl.co/cart/${userID}`, { product: { _id: id } });
+      if (response.status === 200) {
+        dataDispatch({ type: 'ADD_TO_CART', payload: wishlistItem });
+      }
+    } catch (error) {
+      console.error('Error adding to cart ', error);
+    }
+  };
+
+  const toggleFavourite = async (id) => {
+    try {
+      let response;
+      const userID = 124;
+
+      if (isInWishlist) {
+        response = await axios.delete(`https://shoemeup.pranshudobhal.repl.co/wishlist/${userID}/${id}`);
+      } else {
+        response = await axios.post(`https://shoemeup.pranshudobhal.repl.co/wishlist/${userID}`, { product: { _id: id } });
+      }
+
+      if (response.status === 200) {
+        dataDispatch({ type: 'TOGGLE_FAVOURITE', payload: wishlistItem });
+      }
+    } catch (error) {
+      console.error('Error adding to wishlist ', error);
+    }
+  };
```

Comment:
You can move this async calls to app.js or  context to use those functions across the other component instead of writing it repeatedly.

Reply:
Thanks for this suggestion. I think this will significantly reduce my lines of code as I had been calling same functions in various pages.